src/sentry/integrations/utils/metrics.py

```diff
@@ -93,6 +93,50 @@ def get_metric_tags(self) -> Mapping[str, str]:
         }
 
 
+@contextmanager
+def event_lifecycle_context(
+    payload: EventLifecycleMetric,
+    assume_success: bool = True,
+    sample_log_rate: float = 1.0,
+) -> Generator[EventLifecycle]:
+    """
+    Context manager for measuring event lifecycle.
+
+    Args:
+        payload: The metric to measure.
+        assume_success: Whether to assume success if the context is exited without a failure.
+        sample_log_rate: The rate at which to sample logs.
+    """
+    lifecycle = EventLifecycle(payload, assume_success, sample_log_rate)
+
+    # Enter logic
+    if lifecycle._state is not None:
+        lifecycle._report_flow_error("The lifecycle has already been entered")
+    lifecycle._state = EventLifecycleOutcome.STARTED
```

Comment:
My initial impression is that moving the context manager to be outside the class feels weird since we're also accessing 'private' fields that are very specific to `EventLifecycle`. Additionally because EventLifecycle is a class that also keeps track of a state I think it would be better to keep all logic + operations in the same place including the context manager despite the more verbose __enter__ and __exit__ implementations (hopefully this made some sense ? ;-;)